bulk_data_gen/metaqueries/generate_data.go

```diff
@@ -0,0 +1,115 @@
+package metaqueries
+
+import (
+	"time"
+
+	"math/rand"
+
+	"github.com/influxdata/influxdb-comparisons/bulk_data_gen/common"
+)
+
+// letters are used for generating the pseudo-random tag values used in this
+// dataset. Currently only the first 10 letters are used.
+const letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+var (
+	TagKeys = [][]byte{
+		[]byte("X"),
+		[]byte("Y"),
+	}
+	valKey  = []byte("val")
+	measKey = []byte("example_measurement")
+)
+
+type MetaquerySimulatorConfig struct {
+	Start time.Time
+	End   time.Time
+
+	ScaleFactor int
+}
+
+func (d *MetaquerySimulatorConfig) ToSimulator() *MetaquerySimulator {
+	dg := &MetaquerySimulator{
+		madePoints: 0,
+		madeValues: 0,
+		maxPoints:  int64(d.ScaleFactor) * int64(d.ScaleFactor),
+
+		axis:    d.ScaleFactor,
+		TagList: make(map[int][]byte),
+
+		timestampNow:   d.Start,
+		timestampStart: d.Start,
+		timestampEnd:   d.End,
+	}
+
+	// Tag values are generated and stored in memory for later use. The number of
+	// tag values is equal to the scaleFactor. The same list of tag values is used
+	// for both the "X" and "Y" tag keys.
+	for i := 0; i < d.ScaleFactor; i++ {
+		seed := rand.Perm(10)
+		thisVal := make([]byte, 10)
+
+		for idx, v := range seed {
+			thisVal[idx] = letters[v]
+		}
+
+		dg.TagList[i] = thisVal
+	}
+
+	// The stepTime is the interval between generated points. This will result in
+	// the generated data being spread evenly across the provided time range.
+	dg.stepTime = time.Duration(int64(dg.timestampEnd.Sub(dg.timestampStart)) / dg.maxPoints)
+
+	return dg
+}
+
+type MetaquerySimulator struct {
+	madePoints int64
+	maxPoints  int64
+	madeValues int64
+
+	axis    int
+	TagList map[int][]byte
+
+	timestampNow   time.Time
+	timestampStart time.Time
+	timestampEnd   time.Time
+	stepTime       time.Duration
+}
+
+func (g *MetaquerySimulator) SeenPoints() int64 {
```

Comment:
Any particular reason the names of these functions don't match the member names or aren't named as getters?

Reply:
There's an existing interface `Simulator` that it needs to satisfy: https://github.com/influxdata/influxdb-comparisons/blob/master/bulk_data_gen/common/simulation.go#L17-L23

I'll add a comment about that so it's more clear.